### ISSUE TITLE
Enhancements to overlayWith method

### ISSUE THREAD

**Issue by lovell** [lovell/sharp#239]
The main aim here is to add support for Buffer input and allow for differing dimensions with an optional `gravity`. The API should look something like:

`withOverlay(image, options)`

where:

* `image` is a String filepath or Buffer image data
* optional `options` is an Object containing `gravity` e.g. `sharp.gravity.southeast`, defaults to `sharp.gravity.centre`

```javascript
sharp('under.jpg')
  .overlayWith('over.png', {
    gravity: sharp.gravity.northwest
  })
  .toBuffer()
  .then( function(data) {
      ...
  });
```

Resizing of the image to overlay can be carried out beforehand using the existing resize method.

Applying a padding to the image to overlay will be possible with #128. This too can be carried out beforehand.

(See #97 for the original work on the `overlayWith` feature.)

**Comment by fentas** [comment]
I am wondering if there is any progress on this?
Especially of using an Buffer (text as watermark).

**Comment by lovell** [comment]
@fentas I expect these enhancements to be much easier/faster to implement after #152.

**Comment by lovell** [comment]
Commit d92ea31 on the needle/v0.14.0 branch enhances the existing `overlayWith` method by adding support for Buffer-based input, allowing differing formats (8-bit, 16-bit), allowing differing image sizes and therefore accepting a `gravity` option.

This can be tested via:
```sh
npm install lovell/sharp#needle
```

Many thanks to @chrisriley for his original PR #335 on which I based this work.

**Comment by lovell** [comment]
v0.14.0 now available via npm, thanks again for all the help with this.

**Comment by lemnisk8** [comment]
@lovell Please expose the replicate feature from Vips 
replicate( int across , int down )
It would be really useful to tile watermark images for overlay

**Comment by lovell** [comment]
@lemnisk8 Please can you create a new issue for this suggestion along with some examples of how you think the API might look, so we can invite others to discuss and/or submit a PR.
